mvcc/kvstore_txn.go

```diff
@@ -134,6 +158,15 @@ func (tr *storeTxnRead) rangeKeys(key, end []byte, curRev int64, ro RangeOptions
 		limit = len(revpairs)
 	}
 
+	if limit > expensiveReadLimit && !tr.txlocked && tr.ro { // first expensive read in a read only transcation
+		// too many keys to range. upgrade the read transcation to concurrent read tx.
```

Comment:
As an alternative to upgrading the `tr.tx`, would it make sense to have both a `tr.tx` and a `tr.committedTx`? This would limit what is put in the committed read transaction to only expensive reads.

Reply:
nvm, once we perform a `CommittedReadTx` there is no point having two transactions.